src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -3485,6 +3486,44 @@ internal void SymbolDeclaredEvent(Symbol symbol)
             EventQueue?.TryEnqueue(new SymbolDeclaredCompilationEvent(this, symbol.GetPublicSymbol()));
         }
 
+        internal override void SerializePdbEmbeddedCompilationOptions(BlobBuilder builder)
+        {
+            WriteValue("checked", Options.CheckOverflow.ToString());
+            WriteValue("nullable", Options.NullableContextOptions.ToString());
+            WriteValue("unsafe", Options.AllowUnsafe.ToString());
+            WriteValue("langversion", LanguageVersion.ToDisplayString());
+
+            var isOptimized = Options.OptimizationLevel == OptimizationLevel.Release ? true : false;
+            WriteValue("optimize", isOptimized.ToString());
+
+            var preprocessorSymbols = GetPreprocessorSymbols();
+            if (preprocessorSymbols.Any())
+            {
+                WriteValue("define", string.Join(",", preprocessorSymbols));
+            }
+
+            void WriteValue(string key, string value)
+            {
+                builder.WriteUTF8(key);
+                builder.WriteByte(0);
+                builder.WriteUTF8(value);
+                builder.WriteByte(0);
+            }
+        }
+
+        private ImmutableArray<string> GetPreprocessorSymbols()
+        {
+            var firstTree = SyntaxTrees
+                .FirstOrDefault(t => t is CSharpSyntaxTree) as CSharpSyntaxTree;
```

Comment:
Hmm ... do all C# trees have the same symbols? I thought there were cases, particularly in VB, where the symbols differed. @agocke 

@chsienki is this properly handled for SG?

Reply:
I believe compilations created by command line compiler or msbuild task have the same symbols. SyntaxTrees created via API can differ in symbols, but we are not aiming to support arbitrary compilations created via API for this feature.

---
In reply to: [433400402](https://github.com/dotnet/roslyn/pull/44373#discussion_r433400402) [](ancestors = 433400402)

Reply:
See (VisualBasicCompiler|CSharpCompiler).CreateCompilation: the same parse options are used for all trees.

---
In reply to: [434260003](https://github.com/dotnet/roslyn/pull/44373#discussion_r434260003) [](ancestors = 434260003,433400402)

Reply:
It'd be good to add a comment.

---
In reply to: [434260526](https://github.com/dotnet/roslyn/pull/44373#discussion_r434260526) [](ancestors = 434260526,434260003,433400402)

Reply:
In VB the same parse options are set to CompilationOptions.ParseOptions as they are passed to the syntax trees.

---
In reply to: [434260724](https://github.com/dotnet/roslyn/pull/44373#discussion_r434260724) [](ancestors = 434260724,434260526,434260003,433400402)